### PR TITLE
Numeric input tests

### PR DESCRIPTION
**Pull request by johnbender**
Tests! 

**NOTE** There is also a fix for a bug that I thought of when I was testing the `onPaste` method. Namely, it wasn't set up to handle negative or decimal pastes.

### CODE REVIEW

File: test/numeric-input.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+	test("onPaste", function(){
+		// one for prevent default, one for the check
+		expect(14);
+		testOnPaste("123", "123");
+		testOnPaste("-1.23", "-1.23");
+		testOnPaste("12a3", "123");
+		testOnPaste("-123", "-123");
+		testOnPaste("-1-23", "-123");
+		testOnPaste("-1.2.-3", "-1.23");
+		testOnPaste("-.2.-3", "-.23");
```
Comment (johnbender) [inline test/numeric-input.js]: Note the last three tests here outline how pasted values with negatives and decimals will be handled.

File: js/numeric-input.js [hunk 2]
```diff
@@ -2,10 +2,19 @@
...
+
+		// prevent double init
+		if( this.$el.data( this.constructor.name ) ){
+			return;
+		}
+
+		this.$el.data( this.constructor.name, this);
+
+		this.allowFloat =
+			this.$el.is( '[data-float]' ) || this.$el.is( '[data-numeric-input-float]' );
```
Comment (johnbender) [inline js/numeric-input.js]: prevent double init and make the instance available in the super secret data hiding place

File: js/numeric-input.js [hunk 3]
```diff
@@ -139,7 +149,15 @@
...
-		this.el.value = pastedText.replace(/[^0-9\.,]*/g, "");
+		this.el.value = pastedText
+			// remove signs that appear inside the string
+			.replace(/(.+)\-(.+)/g, "$1$2")
+
+			// remove all decimals beside the first
+			.replace(/(.+[\.,].+)\.(.+)/g, "$1$2")
+
+			// remove any non float/integer characters left
+			.replace(/[^0-9\.\-,]*/g, "");
```
Comment (johnbender) [inline js/numeric-input.js]: fixes for pasted values with negative and decimal notation

File: js/numeric-input.js (hunk 2 above)
Comment (zachleat) [inline js/numeric-input.js]: @johnbender Any information on how well `constructor.name` is supported? What ECMAScript version is that from?

**Comment by zachleat** [comment]
👍

**Pull request merged**